lib/icinga/checkable-check.cpp

```diff
@@ -242,6 +243,17 @@ void Checkable::ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrig
 			OnReachabilityChanged(this, cr, children, origin);
 	}
 
+	if (recovery) {
+		// W/o this children could recover too late, after their problem notifications have been sent.
+		for (auto& child : children) {
+			if (child->GetProblem() && child->GetEnableActiveChecks() && !child->IsLikelyToBeCheckedSoon()) {
+				ObjectLock oLock (child);
+				// Rationale for +10...: Ensure this method has enough time to finish.
+				child->SetNextCheck(now + 10 + rand() % 40);
```

Comment:
This should use `Utility::Random()` which provides a thread-safe wrapper using `rand_r()`.